src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,96 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [description, setDescription] = useState('');
+  const [title, setTitle] = useState('');
+  const [imgUrl, setImgUrle] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const isEror = (
+    !title.trim()
+    || !imgUrl.trim()
+    || !imdbId.trim()
+    || !imdbUrl.trim()
+  );
+
+  const resetForm = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrle('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  const handleSabmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setCount(prevCount => prevCount + 1);
+
+    onAdd({
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    });
+
+    resetForm();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={handleSabmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(value) => setTitle(value)}
         required
       />
 
       <TextField
         name="description"
         label="Description"
-        value=""
+        value={description}
+        onChange={(value) => setDescription(value)}
```

Comment:
try to rewrite in all TextField this way 👇
```suggestion
        onChange={setDescription}
```

Just try, if you would have some errors - add screen here and push the way you have 😉